vmware-base/src/main/java/com/cloud/hypervisor/vmware/util/ContentLibraryHelper.java

```diff
@@ -0,0 +1,347 @@
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package com.cloud.hypervisor.vmware.util;
+
+import org.apache.log4j.Logger;
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+
+import com.vmware.vim25.ManagedObjectReference;
+
+import com.vmware.content.library.ItemModel;
+import com.vmware.content.LibraryModel;
+import com.vmware.content.LibraryTypes;
+import com.vmware.content.library.ItemTypes;
+import com.vmware.content.library.StorageBacking;
+import com.vmware.content.library.item.TransferEndpoint;
+import com.vmware.content.library.item.UpdateSessionModel;
+
+import com.vmware.vapi.std.LocalizableMessage;
+
+import com.vmware.vcenter.Datastore;
+import com.vmware.vcenter.DatastoreTypes;
+import com.vmware.vcenter.ovf.OvfError;
+import com.vmware.vcenter.ovf.OvfInfo;
+import com.vmware.vcenter.ovf.OvfMessage;
+import com.vmware.vcenter.ovf.OvfWarning;
+import com.vmware.vcenter.ovf.ParseIssue;
+import com.vmware.vcenter.ovf.LibraryItemTypes;
+import com.vmware.vcenter.ovf.LibraryItemTypes.DeploymentResult;
+import com.vmware.vcenter.ovf.LibraryItemTypes.DeploymentTarget;
+import com.vmware.vcenter.ovf.LibraryItemTypes.ResourcePoolDeploymentSpec;
+
+public class ContentLibraryHelper {
+    private static final Logger s_logger = Logger.getLogger(ContentLibraryHelper.class);
+    private static final String HEADING_ADDITIONAL_INFO = "Additional information :";
+
+    public static boolean createContentLibrary(VmwareContext context, String datastoreName, String libraryName) throws Exception {
+        if (getContentLibraryByName(context, libraryName) != null) {
+            s_logger.debug("Failed to create, content library with the given name: " + libraryName + " already exists");
+            return false;
+        }
+
+        // Build the storage backing for the library to be created
+        StorageBacking storageBacking = createStorageBacking(context, datastoreName);
+
+        // Build the specification for the library to be created
+        LibraryModel createSpec = new LibraryModel();
+        createSpec.setName(libraryName);
+        createSpec.setDescription("Local content library for datastore " + datastoreName);
+        createSpec.setType(LibraryModel.LibraryType.LOCAL);
+        createSpec.setStorageBackings(Collections.singletonList(storageBacking));
+
```

Comment:
sounds like a method should be create called `buildTheSpecificationForTheLibraryToBeCreated(...)`